bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -91,66 +91,12 @@ public String getPrimaryIpv4HostAddress() {
      */
     @Deprecated
-    public static String getLocalIpv4HostAddress() {
-        try {
-            String hostAddress = null;
-            final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
-            while (interfaces.hasMoreElements()) {
-                final NetworkInterface current = interfaces.nextElement();
-                if (!current.isUp() || current.isLoopback() || current.isVirtual()) {
-                    continue;
-                }
-                final Enumeration<InetAddress> addresses = current.getInetAddresses();
-                while (addresses.hasMoreElements()) {
-                    final InetAddress current_addr = addresses.nextElement();
-                    if (current_addr.isLoopbackAddress() || (current_addr instanceof Inet6Address)) {
-                        continue;
-                    }
-                    if (hostAddress != null) {
-                        LOGGER.warn("Found multiple local interfaces - ignoring {}", current_addr.getHostAddress());
-                    } else {
-                        hostAddress = current_addr.getHostAddress();
-                    }
-                }
-            }
-            return hostAddress;
-        } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
-            return null;
-        }
-    }
-
-    private String getFirstLocalIPv4Address() {
-        try {
-            String hostAddress = null;
-            final Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
-            while (interfaces.hasMoreElements()) {
-                final NetworkInterface current = interfaces.nextElement();
-                if (!current.isUp() || current.isLoopback() || current.isVirtual()) {
-                    continue;
-                }
-                final Enumeration<InetAddress> addresses = current.getInetAddresses();
-                while (addresses.hasMoreElements()) {
-                    final InetAddress current_addr = addresses.nextElement();
-                    if (current_addr.isLoopbackAddress() || (current_addr instanceof Inet6Address)) {
-                        continue;
-                    }
-                    if (hostAddress != null) {
-                        LOGGER.warn("Found multiple local interfaces - ignoring {}", current_addr.getHostAddress());
-                    } else {
-                        hostAddress = current_addr.getHostAddress();
-                    }
-                }
-            }
-            return hostAddress;
-        } catch (SocketException ex) {
-            LOGGER.error("Could not retrieve network interface: {}", ex.getMessage(), ex);
-            return null;
-        }
+    public static @Nullable String getLocalIpv4HostAddress() {
+        return new NetUtil().getPrimaryIpv4HostAddress();
```

Comment:
In this case, the user's configures `primaryAddress` will never be taken into account. This might not meet expectations...

Reply:
This is why this method is deprecated. It is a static method, we have no bundle configuration, so the user selected address is not available.